src/components/content.js

```diff
@@ -698,15 +699,27 @@ class Content extends React.Component {
     if (this.props.readOnly) return
     if (!this.isInEditor(event.target)) return
 
-    event.preventDefault()
-    const data = getTransferData(event.clipboardData)
+    const handleData = (data) => {
```

Comment:
No problem. In what way would you call `this.props.onPaste`:

- from a callback given to the utility function like this
`getHtmlFromNativePaste(this, data, dataWithHtml => this.props.onPaste(event, dataWithHtml))` 

- or slightly different `getHtmlFromNativePaste(this, htmlData => this.props.onPaste(event, { ...data, htmlData }))` with `callback({ html, type: 'html' })` from inside the utility function

- or `getHtmlFromNativePaste(this, event, data)` and calling directly from inside the utility function `this.props.onPaste(event, { ...data, html, type: 'html' })`

The first two versions still use a 'proxy' callback, in the last version, it's not obvious that there is an asynchronous handling without looking into the utility function.